python-spec/src/somacore/data.py

```diff
@@ -46,15 +46,26 @@ def create(
         platform_config: Optional[options.PlatformConfig] = None,
         context: Optional[Any] = None,
     ) -> Self:
-        """Creates a new DataFrame."""
+        """Creates a new ``DataFrame`` at the given URI.
+
+        :param uri: The URI where the ``DataFrame`` will be created.
+        :param schema: Arrow schema defining the per-column schema. This schema
+            must define all columns, including columns to be named as index
+            columns.  If the schema includes types unsupported by the SOMA
+            implementation, an error will be raised.
+        :param index_column_names: A list of column names to use as user-defined
+            index columns (e.g., ``['cell_type', 'tissue_type']``).
+            All named columns must exist in the schema, and at least one
+            index column name is required.
+        """
```

Comment:
quite a few methods are missing lifecycle tags

Reply:
added to all methods